watcher/incident_registry.py

```diff
@@ -0,0 +1,41 @@
+from dataclasses import dataclass
+from typing import Optional
+
+
+@dataclass
+class Incident:
+    component: str
+    description: str
+    impact: str
+    status: str
+    timestamp: float
+    notified: bool
+
+
+class IncidentRegistry:
+
+    def __init__(self):
+        self.registry = []
+
+    def add(self, incident: Incident):
+        self.registry.append(incident)
+
+    def delete(self, incident: Incident):
+        raise NotImplemented
+
+    def list(self) -> list:
+        return self.registry
+
+    def clear(self):
+        self.registry.clear()
+
+    def get(self, component_name: str) -> Optional[Incident]:
+        for incident in self.registry:
+            if incident.component == component_name:
+                return incident
+        return None
+
+    def change_notification(self, component_name: str, value: bool):
```

Comment:
Shouldn't it be named `update`?

Reply:
Fixed

Reply:
I meant just 'update'.

Reply:
Also, it would be good to support updating not only one particular field, but updating a few fields at once.

Reply:
Fixed